docs/cat.rst

```diff
@@ -0,0 +1,13 @@
+
+Cat Layer(experimental)
+===========================
+
+The cat types and functions.
+To enable this, define MIOPEN_BETA_API before including miopen.h.
+
+
+miopenCatForward
```

Comment:
Perhaps a few lines describing what the operation is would be useful.

Reply:
I added the following sentence to the cat doc.
```Concatenates the given sequence of seq tensors in the given dimension.```

Reply:
If possible, please add an example as well, that would help others to understand the use for the API.

Reply:
I've made a sample code, (I haven't applied it to the doc yet.)
if kept short, it same the arguments list, 
and if expanded, most of the content pertains to tensor creation.
Shall I refine and apply the following content?
Or please let me know of any other op's examples for referencing.

```
    const int input_count = 2;
    const int input_shape_size = 4;
    miopenHandle_t handle;
    int dim = 1;
    int input_shape[input_count][input_shape_size] = {{32,128,128,128}, {32,64,128,128}};
    int output_shape[input_shape_size] = {32,192,128,128};
    void* input_gpu_buf[2];
    float* input_buf[2];
    void* output_gpu_buf;
    float* output_buf;
    int input_size[2] = {1, 1};
    int output_size = 0;
    float* input0, *input1, *output;
    bool is_different = false;

    miopenTensorDescriptor_t inputDescs[input_count];
    miopenTensorDescriptor_t outputDesc;

    for (int i = 0 ; i < input_shape_size ; i++)
    {
        input_size[0] *= input_shape[0][i];
        input_size[1] *= input_shape[1][i];
    }
    output_size = input_size[0] + input_size[1];

    input_buf[0] = (float*)malloc(input_size[0] * sizeof(float));
    input_buf[1] = (float*)malloc(input_size[1] * sizeof(float));
    output_buf = (float*)malloc(output_size * sizeof(float));

    for (int i = 0 ; i < input_size[0] ; i++)
        input_buf[0][i] = 1.1 * i;

    for (int i = 0 ; i < input_size[1] ; i++)
        input_buf[1][i] = 2.2 * i;

    memset(output_buf, 0, output_size * sizeof(float));

    miopenCreate(&handle);

    miopenCreateTensorDescriptor(&outputDesc);
    miopenCreateTensorDescriptor(&inputDescs[0]);
    miopenCreateTensorDescriptor(&inputDescs[1]);

    miopenSetTensorDescriptor(inputDescs[0],
                              miopenFloat,
                              4,
                              input_shape[0],
                              nullptr);

    miopenSetTensorDescriptor(inputDescs[1],
                              miopenFloat,
                              4,
                              input_shape[1],
                              nullptr);

    miopenSetTensorDescriptor(outputDesc,
                              miopenFloat,
                              4,
                              output_shape,
                              nullptr);

    hipMalloc(&input_gpu_buf[0], input_size[0] * sizeof(float));
    hipMalloc(&input_gpu_buf[1], input_size[1] * sizeof(float));
    hipMalloc(&output_gpu_buf, output_size * sizeof(float));

    hipMemcpy(input_gpu_buf[0], input_buf[0], input_size[0] * sizeof(float), hipMemcpyHostToDevice);
    hipMemcpy(input_gpu_buf[1], input_buf[1], input_size[1] * sizeof(float), hipMemcpyHostToDevice);

    miopenCatForward(handle,
                     2,
                     inputDescs,
                     input_gpu_buf,
                     outputDesc,
                     output_gpu_buf,
                     dim);

    hipMemcpy(output_buf, output_gpu_buf, output_size * sizeof(float), hipMemcpyDeviceToHost);

    hipFree(input_gpu_buf[0]);
    hipFree(input_gpu_buf[1]);
    hipFree(output_gpu_buf);

    miopenDestroy(handle);

    input0 = input_buf[0];
    input1 = input_buf[1];
    output = output_buf;

    for (int i = 0 ; i < output_shape[0] ; i++)
    {
        for (int j = 0 ; j < input_size[0]/output_shape[0] ; j++)
        {
            if (*output != *input0)
                is_different = true;
            input0++;
            output++;
	}
        for (int j = 0 ; j < input_size[1]/output_shape[0] ; j++)
        {
            if (*output != *input1)
                is_different = true;
            input1++;
            output++;
        }
    }

    free(input_buf[0]);
    free(input_buf[1]);
    free(output_buf);

    if (is_different)
        printf("The input data and output data are distinct from each other.\n");
    else
        printf("The input data and output data are identical.\n");
```